server/src/domain/person/person.service.ts

```diff
@@ -63,8 +63,8 @@ export class PersonService {
   async update(authUser: AuthUserDto, id: string, dto: PersonUpdateDto): Promise<PersonResponseDto> {
     let person = await this.findOrFail(authUser, id);
 
-    if (dto.name != undefined || dto.isHidden !== undefined) {
-      person = await this.repository.update({ id, name: dto.name, isHidden: dto.isHidden });
+    if (dto.name != undefined || dto.birthDate !== undefined || dto.isHidden !== undefined) {
+      person = await this.repository.update({ id, name: dto.name, birthDate: dto.birthDate, isHidden: dto.isHidden });
```

Comment:
This is only here, really, to determine when `SEARCH_INDEX_ASSET` needs to run. It is quite an expensive operation (loading every asset for a person and re-indexing it). It only needs to run when the name or visibility of the people changes. This is not the case for person birthdays.

Can we change it so the `await this.repository.update` always runs and the `SEARCH_INDEX_ASSET` part only runs if `name` or `isHidden` are updated?